src/freenet/node/NewPacketFormat.java

```diff
@@ -398,7 +398,7 @@ private NPFPacket decipherFromSeqnum(byte[] buf, int offset, int length, Session
 		byte[] payload = Arrays.copyOfRange(buf, offset + hmacLength, offset + length);
 		byte[] hash = Arrays.copyOfRange(buf, offset, offset + hmacLength);
 
-		if(!HMAC.verifyWithSHA256(sessionKey.hmacKey, payload, hash)) return null;
+		if(!HMAC.verifyWithSHA256(sessionKey.hmacKey, payload, hash, HMAC_LENGTH)) return null;
```

Comment:
The rest of the code uses `hmacLength`, including the lines above this one. Please use that one instead.